extensions/flyway/runtime/src/main/java/io/quarkus/flyway/runtime/FlywayInitTask.java

```diff
@@ -0,0 +1,47 @@
+package io.quarkus.flyway.runtime;
+
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.enterprise.inject.Any;
+import jakarta.enterprise.inject.Instance;
+import jakarta.inject.Inject;
+
+import org.eclipse.microprofile.config.inject.ConfigProperty;
+import org.flywaydb.core.Flyway;
+
+import io.quarkus.runtime.annotations.PreStart;
+
+@ApplicationScoped
+@PreStart("flyway-init-task")
```

Comment:
At the moment we do not discern between init jobs and therefore your init task code should handle that, until we add support for soemthing like that.

Two ways I can think of: 

1. Only run when `QUARKUS_INIT_AND_EXIT` = `true` which is a condition that will be satisfied in the Job container.
2. Check the hostname of the container in order to check that its an init jobs (has the init suffix).

Moving forward we could possibly add additional enviornment variables to the Job so that its easier for the task to detect the execution context.

Reply:
Added additional env varialbe so that its clear when a container is a a job container.
Also added env var that specifies the init task name.

Reply:
Hum, I'm confused... so basically, you are saying that all extensions must check if they are in an initialization job and if that's their own job; otherwise, do nothing? That's a huge requirement, and now we need to add that to a lot of extension (scheduler for example, or it will start doing stuff).

Reply:
(even worse for kafka where I would have a bunch of extension that would need to check things that do not belong to them)